lib/core/job_cleanup.rb

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'core'
+require 'utility/logger'
+
+module Core
+  class JobCleanUp
```

Comment:
Would it be possible to change the class name to `CleanUpJob` to follow the convention where the main noun is at the end of the name like: 

```
ConnectorJob
ElasticConnectorActions
Utility::Logger
SyncJobRunner
```

Another idea is to use `Jobs` namespaces that I created, in that case, it can look like `Core::Jobs::CleanUpJob`. I think it's a good place for all jobs from my point of view.

Reply:
I'm not a fan of using `verb` as a class name. What do you think of `JobCleaner`?

Reply:
CleanUp is a "noun as adjective" here IMO

Reply:
What my previous comment meant is "I like CleanupJob" a bit more, but it's opinion

Reply:
Oh, I didn't notice this. I think at least @vidok and @artem-shelkovnikov are voting `CeanUpJob`, I'm not sure about the 👍 from @seanstory , but I'm going to rename it.

I also feel it makes sense to put it under the module `Jobs`. Should we name it `Cleaner`, as consistent with `Producer` and `Consumer`? Or stick with `CleanUp` (or `Cleanup`, so the file name of `clean_up.rb` vs `cleanup.rb`)?